common/djangoapps/edxmako/management/commands/preprocess_assets.py

```diff
@@ -8,67 +8,53 @@
 
-    def handle_noargs(self, **options):
-        """
-        Walk over all of the static files directories specified in the
-        settings file, looking for asset template files (indicated by
-        a file extension like .mako).
-        """
-        for staticfiles_dir in getattr(settings, "STATICFILES_DIRS", []):
-            # Cribbed from the django-staticfiles app at:
-            # https://github.com/jezdez/django-staticfiles/blob/develop/staticfiles/finders.py#L52
-            if isinstance(staticfiles_dir, (list, tuple)):
-                prefix, staticfiles_dir = staticfiles_dir
-
-            # Walk over the current static files directory tree,
-            # preprocessing files that have a template extension.
-            for root, dirs, files in os.walk(staticfiles_dir):
-                for filename in files:
-                    outfile, extension = os.path.splitext(filename)
-                    # We currently only handle Mako templates
-                    if extension == ".mako":
-                        self.__preprocess(os.path.join(root, filename),
-                                          os.path.join(root, outfile))
-
-    def __context(self):
-        """
-        Return a dict that contains all of the available context
-        variables to the asset template.
-        """
-        # TODO: do we need to include anything else?
-        # TODO: do this with the django-settings-context-processor
-        return {
-            "FEATURES": settings.FEATURES,
-            "THEME_NAME": getattr(settings, "THEME_NAME", None),
-        }
-
-    def __preprocess(self, infile, outfile):
-        """
-        Run `infile` through the Mako template engine, storing the
-        result in `outfile`.
-        """
-        with open(outfile, "w") as _outfile:
-            _outfile.write(textwrap.dedent("""\
-            /*
-             * This file is dynamically generated and ignored by Git.
-             * DO NOT MAKE CHANGES HERE. Instead, go edit its template:
-             * %s
-             */
-            """ % infile))
-            _outfile.write(Template(filename=str(infile)).render(env=self.__context()))
+    def handle(self, *args, **options):
+        theme_name = getattr(settings, "THEME_NAME", None)
+        use_custom_theme = settings.FEATURES.get("USE_CUSTOM_THEME", False)
+        if not use_custom_theme or not theme_name:
+            # No custom theme, nothing to do!
+            return
+
+        dest_dir = args[-1]
+        for source_file in args[:-1]:
+            self.process_one_file(source_file, dest_dir, theme_name)
+
+    def process_one_file(self, source_file, dest_dir, theme_name):
+        """Pre-process a .scss file to replace our markers with real code."""
+        with open(source_file) as fsource:
+            original_content = content = fsource.read()
+
+        content = content.replace(
+            "//<THEME-OVERRIDE>",
+            "@import '{}';".format(theme_name),
```

Comment:
Stanford theming depends on having this conditional import. Previously (which is to say, in the current codebase), it was achieved by pre-processing the Sass files first through Mako: [here's an example](https://github.com/edx/edx-platform/blob/ebed8cd42ab0f7a07855061afc393109c9572769/lms/static/sass/lms-course.scss.mako#L9-L20). This method of replacing a comment with an import seems less dirty to me.

We also would like to rip out Stanford theming at some point in the future, so that we don't need to do this conditional import at all. However, we can't do that in this pull request -- there has to be some overlap of supporting both Stanford theming and this new comprehensive theming system, so that people who are currently using Stanford theming have some time to migrate over to the new system.